src/commands/deployWorkspaceProject/internal/startingConfiguration/DwpManagedEnvironmentListStep.ts

```diff
@@ -66,4 +74,45 @@ export class DwpManagedEnvironmentListStep extends AzureWizardPromptStep<DeployW
         context.resourceGroup = resourceGroups.find(rg => rg.name === getResourceGroupFromId(nonNullProp(managedEnvironment, 'id')));
         context.managedEnvironment = managedEnvironment;
     }
+
+    private async setRecommendedPicks(context: DeployWorkspaceProjectInternalContext, picks: IAzureQuickPickItem<ManagedEnvironment | undefined>[]): Promise<void> {
+        const deploymentConfigurations: DeploymentConfigurationSettings[] | undefined = await dwpSettingUtilsV2.getWorkspaceDeploymentConfigurations(nonNullProp(context, 'rootFolder'));
+        if (!deploymentConfigurations?.length) {
+            return;
+        }
+
+        const client = await createContainerAppsAPIClient(context);
+        for (const config of deploymentConfigurations) {
+            try {
+                if (config.resourceGroup && config.containerApp) {
+                    const containerApp = await client.containerApps.get(config.resourceGroup, config.containerApp);
+                    const recommendedPick = picks.find(p => p.data?.id === containerApp.managedEnvironmentId);
+                    if (recommendedPick) {
+                        recommendedPick.description = recommendedPickDescription;
+                    }
+                }
+            }
+            catch (these_hands) {
+                // ignore the error and continue
+            }
+        }
+
+        const recommendedPick = (pick: IAzureQuickPickItem<ManagedEnvironment | undefined>): boolean => {
```

Comment:
This has the same name as the `recommendedPick` variable int he if block.  I know that version of the variable is block scope, but maybe a different name here is a good idea :)

Also, nit but a slightly shorter alternative:

```typescript
const recommendedPick = (pick: IAzureQuickPickItem<ManagedEnvironment | undefined>): boolean => pick.description === recommendedPickDescription;
```

Reply:
Good call. I was doing a counter to keep track of how many were recommended before which is why I had the if/else.